qemu/tests/boot_from_device.py

```diff
@@ -169,7 +178,8 @@ def check_boot_result(boot_entry_info, boot_fail_info, device_name):
                 vm.send_key(str(i))
                 break
         else:
+            cleanup(dev_name)
```

Comment:
I think this line and line 100 is not necessary.
Can you please explain why they are needed?

Reply:
@hereischen I suppose they are all necessary, to clean up before fail the test case.